django/db/migrations/state.py

```diff
@@ -602,3 +631,122 @@ def __eq__(self, other):
+
+    @property
+    def model_name(self):
+        return self.model.name_lower
+
+    @cached_property
+    def db_table(self):
+        db_table = self.model.options.get('db_table')
+        if db_table is None:
+            db_table = "%s_%s" % (self.model.app_label, self.model.name_lower)
+        return truncate_name(db_table, connection.ops.max_name_length())
+
+    @cached_property
+    def db_tablespace(self):
+        return self.model.options.get('db_tablespace', None)
+
+    @cached_property
+    def index_together(self):
+        return self.model.options.get('index_together', [])
+
+    @cached_property
+    def indexes(self):
+        return self.model.options.get('indexes', [])
+
+    @cached_property
+    def label(self):
+        return '%s.%s' % (self.model.app_label, self.model.name)
+
+    @cached_property
+    def label_lower(self):
+        return '%s.%s' % (self.model.app_label, self.model.name_lower)
+
+    @cached_property
+    def managed(self):
+        return self.model.options.get('managed', True)
+
+    @cached_property
+    def order_with_respect_to(self):
+        return self.model.options.get('order_with_respect_to', None)
+
+    @cached_property
+    def pk(self):
+        for name in self.model.fields:
+            if self.model.fields[name].primary_key:
+                return self.model.fields[name]
+
+
+
+    @cached_property
+    def proxy(self):
+        return self.model.options.get('proxy', False)
+
+    @cached_property
+    def required_db_features(self):
+        return self.model.options.get('required_db_features', [])
+
+    @cached_property
+    def required_db_vendor(self):
+        return self.model.options.get('required_db_vendor', None)
+
+    @cached_property
+    def swappable(self):
+        return self.model.options.get('swappable', None)
+
+    @cached_property
+    def swapped(self):
+        """
+        Has this model been swapped out for another? If so, return the model
+        name of the replacement; otherwise, return None.
+        For historical reasons, model name lookups using get_model() are
+        case insensitive, so we make sure we are case insensitive here.
+        """
+        if self.swappable:
+            swapped_for = getattr(settings, self.swappable, None)
+            if swapped_for:
+                try:
+                    swapped_label, swapped_object = swapped_for.split('.')
+                except ValueError:
+                    # setting not in the format app_label.model_name
+                    # raising ImproperlyConfigured here causes problems with
+                    # test cleanup code - instead it is raised in get_user_model
+                    # or as part of validation.
+                    return swapped_for
+
+                if '%s.%s' % (swapped_label, swapped_object.lower()) not in (None, self.label_lower):
+                    return swapped_for
+        return None
+
+    @cached_property
+    def unique_together(self):
```

Comment:
UnqiueConstraint is suggested instead of unique_togather in newer versions [2.2+]